OpenRA.Mods.Common/Traits/Infantry/TakeCover.cs

```diff
@@ -38,22 +56,25 @@ public class TakeCover : Turreted, INotifyDamage, IDamageModifier, ISpeedModifie
 		public bool IsModifyingSequence { get { return IsProne; } }
 		public string SequencePrefix { get { return info.ProneSequencePrefix; } }
 
+		readonly HashSet<string> proneTriggers;
+
 		public TakeCover(ActorInitializer init, TakeCoverInfo info)
 			: base(init, info)
 		{
 			this.info = info;
+			proneTriggers = info.ProneTriggers.ToHashSet();
```

Comment:
This feels like premature optimization, and if profiling shows it to make a measurable difference then it should be done to all of these type lists in its own PR.